### ISSUE TITLE
Failed to build on VS 2022 17.0.2 - 'expression did not evaluate to a constant' . in  write.hpp

### ISSUE THREAD

**Issue by akmal-ali** [stephenberry/glaze#3]
Code failed to build using VS 2022. Error is 'expression did not evaluate to a constant' . in  write.hpp. Line 232.
``` cpp
               static constexpr auto member_ptr =
                  std::get<member_it->second.index()>(member_it->second);
```
[{
	"resource": "/C:/Users/akmal/projects/cpp/glaze/include/glaze/binary/write.hpp",
	"owner": "cmake-build-diags",
	"code": "C2131",
	"severity": 8,
	"message": "expression did not evaluate to a constant [C:\\Users\\akmal\\projects\\cpp\\glaze\\build\\_glaze.vcxproj]",
	"source": "MSVC",
	"startLineNumber": 232,
	"startColumn": 49,
	"endLineNumber": 232,
	"endColumn": 49
}]

**Comment by stephenberry** [comment]
Wait, still broken... fixed a bug near this one.

**Comment by stephenberry** [comment]
Okay, this commit fixes the problem (06c0e16). Thanks again!
